File: app/src/main/java/me/rooshi/podcastapp/Post.java
package me.rooshi.podcastapp;

import com.google.firebase.database.IgnoreExtraProperties;

@IgnoreExtraProperties
public class Post {
    public String date;
    public int likes;
    public String text;
    public String user;

    public Post() {

    }

    public Post(String date, int likes, String text, String user) {
        this.date = date;
        this.likes = likes;
        this.text = text;
        this.user  = user;
    }
}

File: domain/src/main/java/me/rooshi/domain/model/Post.kt
package me.rooshi.domain.model

//This class should represent a post and a comment
// If we're doing it like twitter, the parent post has no parent, but "comments" have a parent and
//  we just display them differently if needed. Comments should also be able to link to other clips etc.

//We don't need a like class, just a list of users who liked this post.
// A reference to a liked post should also be put in the user model so they can see what posts they have liked.

data class Post (val id: String,
            val userId: String,
            val date: Long,
            val bodyText: String,
            val likes: List<User>,
            val shares: List<User>,
            val comments: List<Post>,
            val clip: Clip?)